### PR TITLE
cleanup propagated resources

### PR DESCRIPTION
**Pull request by huiwq1990**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/karmada-io/karmada/issues/438

**Special notes for your reviewer**:

1) Similar proposal at kubefed project, https://github.com/kubernetes-sigs/kubefed/pull/1233
2) Currently only provide cluster-scope cleanup, not provide propagated-resource-scope cleanup

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: docs/keps/cleanup-propagated-resources.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+* Define how users can indicate that propagated resource should be removed when a cluster leaves
+* Define cluster removal strategies
+
+## Proposals
+
+The Cluster struct should be updated to contain a `RemoveStrategy` member. `RemoveStrategy` will initially support two values, `Needless` and  `Required`.
+
+- The `Needless` strategy will not cleanup any propagated resources.
+- The `Required` strategy will halt the unjoin process and set the `Cluster resource` in a failed state when encountering errors. Unjoining is blocked until all propagated resources have been removed successfully.
```
Comment (RainbowMango) [inline docs/keps/cleanup-propagated-resources.md:49]: I don't know if there is a better name.

**Comment by RainbowMango** [comment]
Would you like to talk about this at the next meeting?
If yes, remember to add an agenda to the [meeting agenda](https://docs.google.com/document/d/1y6YLVC-v7cmVAdbjedoyR5WL0-q45DBRXTvz5_I7bkA/edit).

**Comment by huiwq1990** [comment]
> meeting agenda

I will joint, and suggest put zoom address in the doc.

**Comment by RainbowMango** [comment]
Hi @huiwq1990 

I helped rebase this PR.
We can merge this proposal for now, and postpone the development work until there is really a need.
Are you ok with that?

**Comment by RainbowMango** [comment]
No response from @huiwq1990 . Let's do it and close this.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**